src/common/logger.c

```diff
@@ -128,19 +130,13 @@ bool is_logger_colored(void) {
   }
 }
 
-pid_t get_tid() {
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wimplicit-function-declaration"
-  return gettid();
-#pragma GCC diagnostic pop
-}
+pid_t get_tid() { return syscall(__NR_gettid); }
```

Comment:
why not using gettid() directly instead of redoing a wrapper which does the same.(for older libc?) (same question from the previous version of the wrapper)

Reply:
Yes the main reason is that `gettid` is not available with older version of linux kernel.

Reply:
Syscall is fine and should be compatible with both old and new versions.

Reply:
Yes, it's the most compatible one. I tested by checking the kernel version but it was not stable across compilers so I went back on `syscall`